### DISCUSSION TITLE
Is there any plan to change the ipam database to SQLite ?

### DISCUSSION THREAD

**Discussion by heyshsh** [containers/podman#23318] [General]
Currently, if using `Netavark` as the network backend, podman would use a boltdb as the ipam database to track the used ips for each container.
`/run/containers/networks/ipam.db`

Is there any plan to change the ipam database from boltdb to SQLite? Thanks.

**Comment by Luap99** [comment]
eventually yes

**Comment by heyshsh** [comment]
Thanks for response. So in which release do you plan to change?

**Reply by Luap99** [reply]
there is no target, which is why I said eventually. Long term it would be nice to drop boltdb but this cannot happen without breaking backwards compatibility which means it has to wait until we want to do that

**Reply by heyshsh** [reply]
Got it, thanks a lot.

**Comment by thomasdba** [comment]
still no plan in 6.* ?

**Reply by Luap99** [reply]
Well I am well aware of this but it simply has not been a priority so far, the main issues with bolt were about data corruption on unclean shutdown. Since ipam is just on tmpfs that is not a real concern there.

Overall long term it would be nice to move this code out of the podman side directly into netavark. So podman just has to call it and not do much complicated logic itself.

If someone likes to propose alternatives they can create design docs https://github.com/podman-container-tools/podman/tree/main/contrib/design-docs and/or talk to me directly about it. I am the main maintainer of the netavark and podman side code which calls netavark (libentwork/netavark).
